### PR TITLE
Fix Shrine overscroll glow indicator

### LINKED ISSUE #26036: Shrine app shows overscroll glow indicator in both dimensions
Take a sufficiently tall Android phone. 

Scrolling horizontally shows an overscroll glow. This is expected.
Scrolling vertically also shows an overscroll glow. This is unexpected since nothing scrolls vertically in this screen.

### PR DESCRIPTION
**Pull request by michdud**
Closes #26036.

I took `ClampingScrollPhysics()` from https://github.com/material-components/material-components-flutter-codelabs and added it to the version of Shrine in the gallery to prevent the vertical overscroll glow indicator (sorry if someone's already got this figured out!).

### CODE REVIEW

**Comment by michdud** [comment]
I signed it!

File: examples/flutter_gallery/lib/demo/shrine/supplemental/product_columns.dart [hunk 1]
```diff
@@ -37,6 +37,7 @@ class TwoProductCardColumn extends StatelessWidget {
         : 33 / 49;
 
       return ListView(
+        physics: ClampingScrollPhysics(),
```
Comment (rami-a) [inline examples/flutter_gallery/lib/demo/shrine/supplemental/product_columns.dart]: Please use `const` for this

File: examples/flutter_gallery/lib/demo/shrine/supplemental/product_columns.dart [hunk 2]
```diff
@@ -71,6 +72,7 @@ class OneProductCardColumn extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     return ListView(
+      physics: ClampingScrollPhysics(),
```
Comment (rami-a) [inline examples/flutter_gallery/lib/demo/shrine/supplemental/product_columns.dart]: Please use `const` for this

**Review by rami-a — COMMENTED** [review]
Thanks for the fix @michdud, if you could fix the analyzer errors that are due to the lack of declaring `ClampingScrollPhysics` as const, then this should be good!

File: examples/flutter_gallery/lib/demo/shrine/supplemental/product_columns.dart (hunk 1 above)
Comment (michdud) [reply examples/flutter_gallery/lib/demo/shrine/supplemental/product_columns.dart]: Done!

File: examples/flutter_gallery/lib/demo/shrine/supplemental/product_columns.dart (hunk 2 above)
Comment (michdud) [reply examples/flutter_gallery/lib/demo/shrine/supplemental/product_columns.dart]: Done!

**Review by rami-a — APPROVED** [review]

**Comment by rami-a** [comment]
Thanks @michdud!

**Comment by willlarche** [comment]
Thank you!

**Pull request merged**